graph/memstore/quadstore.go

```diff
@@ -117,17 +116,17 @@ func (qs *QuadStore) ApplyDeltas(deltas []graph.Delta, ignoreOpts graph.IgnoreOp
 		case graph.Add:
 			if !ignoreOpts.IgnoreDup {
 				if _, exists := qs.indexOf(d.Quad); exists {
-					return graph.ErrQuadExists
+					return &graph.QuadError{Action: d.Action, Quad: d.Quad, Err: graph.ErrQuadExists}
```

Comment:
1) and 2) can be fixed by unexporting the errors that exist (note that no references to these values exist in the code base as it stands) and providing helpers as exist for os: IsQuadExist(error) bool and IsQuadNotExist(error) bool. 3) is addressed by refashioning the error to be a DeltaError (which was what I was really intending).